components/profile-controller/controllers/profile_controller.go

```diff
@@ -710,9 +710,10 @@ func (r *ProfileReconciler) PatchDefaultPluginSpec(ctx context.Context, profileI
 				},
 			})
 		}
-	}
-	if err := r.Update(ctx, profileIns); err != nil {
-		return err
+		if err := r.Update(ctx, profileIns); err != nil {
+			return err
+		}
+
```

Comment:
You should put it even highier - there is no point in updating the profile unless something is changed in this profile. And `profileIns` is changed only in the innermost if statement.

Reply:
Nice catch! Thank you